examples/compgcn/comgcn_trainer.py

```diff
@@ -0,0 +1,177 @@
+# !/usr/bin/env python3
+# -*- coding:utf-8 -*-
+
+# @Time    : 2022/04/20 13:47
+# @Author  : hanhui
+# @FileName: trainer.py.py
+import os
+os.environ['TL_BACKEND'] = 'torch'  # set your backend here, default `tensorflow`
+os.environ["CUDA_VISIBLE_DEVICES"] = ""
+
+import sys
+sys.path.insert(0, os.path.abspath('../../'))  # adds path2gammagl to execute in command line.
+import argparse
+import tensorlayerx as tlx
+from gammagl.models import CompGCN
+from tensorlayerx.model import TrainOneStep, WithLoss
+from gammagl.datasets import Entities
+import os.path as osp
+import numpy as np
+
+
+class SemiSpvzLoss(WithLoss):
+    def __init__(self, net, loss_fn):
+        super(SemiSpvzLoss, self).__init__(backbone=net, loss_fn=loss_fn)
+
+    def forward(self, data, y):
+        logits = self._backbone(data['edge_index'], data['edge_type'])
+        train_logits = tlx.gather(logits, data['train_idx'])
+        # train_y = y # tlx.gather(y, data['train_idx'])
+        loss = self._loss_fn(train_logits, data['train_y'])
+        return loss
+
+
+def evaluate(net, data, y, metrics):
+    net.set_eval()
+    logits = net(data['edge_index'], data['edge_type'])
+    _logits = tlx.gather(logits, data['test_idx'])
+    _y = y # tlx.gather(y, data['test_idx'])
+    metrics.update(_logits, _y)
+    acc = metrics.result()
+    metrics.reset()
+    return acc
+
+def calculate_acc(logits, y, metrics):
+    """
+    Args:
+        logits: node logits
+        y: node labels
+        metrics: tensorlayerx.metrics
+
+    Returns:
+        rst
+    """
+
+    metrics.update(logits, y)
+    rst = metrics.result()
+    metrics.reset()
+    return rst
+
+def main(args):
+    # load cora dataset
+    if str.lower(args.dataset) not in ['aifb', 'mutag', 'bgs', 'am']:
+        raise ValueError('Unknown dataset: {}'.format(args.dataset))
+    path = osp.join(osp.dirname(osp.realpath(__file__)), '../Entities')
+    dataset = Entities(path, args.dataset)
+    graph = dataset[0]
+
+    graph.numpy()
```

Comment:
Why we need to convert to numpy()? In line 71, you convert to tensor()?

Reply:
I have deleted them.